tests/testthat/test-parse_input_data.R

```diff
@@ -9,11 +9,11 @@ test_that("parse_input_data handles regular data frame input", {
   result <- parse_input_data(data)
   
   # Check column names are lowercase
-  expect_equal(names(result), c("t", "obs_type", "y"))
+  expect_equal(names(result), c("t", "obs_type", "y", "obs_type"))
```

Comment:
this is a bit ugly to end up with two columns with same name, but this was how it was working before, we just didn't have a test on it. Don't want to change more than needed in this PR, we can tackle later.